In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [69]:
df = pd.read_csv('/mnt/c/Users/marcn/Documents/Ironhack/sharky/Data/attacks.csv', encoding="ISO-8859-1")
print(df.shape)
display(df.info())
df.sample(3)

(25723, 24)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25723 entries, 0 to 25722
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             8702 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 34

None

Case Number Date  Year Type Country Area Location Activity Name Sex   \
9807          NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
20229         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
11298         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   

       ... Species  Investigator or Source  pdf href formula href  \
9807   ...      NaN                    NaN  NaN          NaN  NaN   
20229  ...      NaN                    NaN  NaN          NaN  NaN   
11298  ...      NaN                    NaN  NaN          NaN  NaN   

      Case Number.1 Case Number.2 original order Unnamed: 22 Unnamed: 23  
9807            NaN           NaN            NaN         NaN         NaN  
20229           NaN           NaN            NaN         NaN         NaN  
11298           NaN           NaN            NaN         NaN         NaN  

[3 rows x 24 columns]

In [76]:
df.columns

Index(['Case Number', 'Date', 'Year', 'Type', 'Country', 'Area', 'Location',
       'Activity', 'Name', 'Sex ', 'Age', 'Injury', 'Fatal (Y/N)', 'Time',
       'Species ', 'Investigator or Source', 'pdf', 'href formula', 'href',
       'Case Number.1', 'Case Number.2', 'original order', 'Unnamed: 22',
       'Unnamed: 23'],
      dtype='object')

We have 23 parameters related to 25723 different cases.
The good news is that we have many cases, the bad news is that the data has a lot of NaNs to be filled. 

We need to filter the data in order to work with it. Decide wether to keep or not each column. 

With the method .unique() we can check the different values of each column, I did so for the columns named 'Unnamed:22' and 'Unamed: 23', and discovered that the data in them is completly irrelevant to the study, so we can drop them. 
'

In [56]:
df.drop(axis=0, index=None, columns=['Unnamed: 23', 'Unnamed: 22'], level=None, inplace=True) 

Another way to select data from the dataframe, is to try and imporve the data in the whole Data frame as it is and then to take from it the columns we are going to use, creating a new data frame insted of dropping data from the original one. 

In [85]:
a = list(df['Country'].unique())
print(a[:20], len(a))


['USA', 'AUSTRALIA', 'MEXICO', 'BRAZIL', 'ENGLAND', 'SOUTH AFRICA', 'THAILAND', 'COSTA RICA', 'MALDIVES', 'BAHAMAS', 'NEW CALEDONIA', 'ECUADOR', 'MALAYSIA', 'LIBYA', nan, 'CUBA', 'MAURITIUS', 'NEW ZEALAND', 'SPAIN', 'SAMOA'] 213


I will try to extract some data from this Data frame. First of all I would like to know how does the number of shark attacks evolve with time. If they have increased or decreased and if they have done so according to the growing rate of earth population. 

I would also like to study where do most of the attacks happen, what countries.

And last but not least, what are the human features that make people look more appetizing to sharks. Is it the Sex/Gender? Is it the Age? Or, maybe, the first letter of their name?


Now, let's hypothesize:
1. First, I think the evolution of Shark attacks during history will rapidly grow from the first ages until now. For two main reasons:
+ In the past most shark attacks weren't reported, so the bast majority of them won't be registered. I assume that we are going to observe a huge growth at some point, wherever we observe a huge slope in the increase of cases is probably where the attacks started to be reported more often. This makes the prior data less reliable, of course. 
+ The second reason that makes it grow is the rapid advance of the population growth in the planet, the more the human activity, the more the sharks are going to attack humans, right?


All in all, I belive that the number of attacks is going to increase with time, first it will increase very rapidly, then it will, probably, 
stabilize, due to the growth on the security and maybe the disminution of shark population, so even though it will still grow it won't do so accordingly to the groth of human population.


2. Where is it more likely that a shark attacks you? My guess, near the tropics. I think is where most of the shark population is placed. In this point of the project it will probably be interesting to study the Species that attack in each place, it will give us an insight on the places where each Specie is more commontly found. 

3. This is, probavly, the easiest part of the study. The ages where humans have more water activities, such as boating or surfing, are going to be the more affected by shark attacks, with a first quick look probably the more affected ages will be between the 20's and the 40's. Probably most of the attacks will be to men. Specially in the past, due to the systematic invisibility of women in society and also because Gener roles make men more reckless (For instance, if we had the data, I am sure that most of the historic cases in the data source are from white men). Lastly, the first letter in the name should be correlated to shark attacks, however we are probably going to find some correlation because there are names that are more common than others. 

